common/metrics/defs.go

```diff
@@ -2196,26 +2207,26 @@ var MetricDefs = map[ServiceIdx]map[int]metricDefinition{
 		PersistenceErrBadRequestCounter:                     {metricName: "persistence_errors_bad_request", metricType: Counter},
 		ClientRequests:                                      {metricName: "client_requests", metricType: Counter},
 		ClientFailures:                                      {metricName: "client_errors", metricType: Counter},
-		ClientLatency:                                       {metricName: "client_latency", metricType: Timer},
+		ClientLatency:                                       {metricName: "client_latency", metricType: Timer, unit: Milliseconds},
 		ClientRedirectionRequests:                           {metricName: "client_redirection_requests", metricType: Counter},
 		ClientRedirectionFailures:                           {metricName: "client_redirection_errors", metricType: Counter},
-		ClientRedirectionLatency:                            {metricName: "client_redirection_latency", metricType: Timer},
-		ServiceAuthorizationLatency:                         {metricName: "service_authorization_latency", metricType: Timer},
-		NamespaceCachePrepareCallbacksLatency:               {metricName: "namespace_cache_prepare_callbacks_latency", metricType: Timer},
-		NamespaceCacheCallbacksLatency:                      {metricName: "namespace_cache_callbacks_latency", metricType: Timer},
-		StateTransitionCount:                                {metricName: "state_transition_count", metricType: Timer},
-		HistorySize:                                         {metricName: "history_size", metricType: Timer},
-		HistoryCount:                                        {metricName: "history_count", metricType: Timer},
-		EventBlobSize:                                       {metricName: "event_blob_size", metricType: Timer},
-		SearchAttributesSize:                                {metricName: "search_attributes_size", metricType: Timer},
+		ClientRedirectionLatency:                            {metricName: "client_redirection_latency", metricType: Timer, unit: Milliseconds},
+		ServiceAuthorizationLatency:                         {metricName: "service_authorization_latency", metricType: Timer, unit: Milliseconds},
+		NamespaceCachePrepareCallbacksLatency:               {metricName: "namespace_cache_prepare_callbacks_latency", metricType: Timer, unit: Milliseconds},
+		NamespaceCacheCallbacksLatency:                      {metricName: "namespace_cache_callbacks_latency", metricType: Timer, unit: Milliseconds},
+		StateTransitionCount:                                {metricName: "state_transition_count", metricType: Histogram, unit: Dimensionless},
+		HistorySize:                                         {metricName: "history_size", metricType: Histogram, unit: Bytes},
```

Comment:
Same feeling here. Maybe we should define a few helper method:
newTimer() // for miliseconds histogram
newBytes() // for bytes histogram
newHistogram() // for dimentionless histogram
newCounter()
newGauge()

Reply:
This is a good suggestion.